File: Lua/bogo_sort.lua
function is_sorted(arr)
    for i = 1, #arr - 1 do
        if arr[i] > arr[i + 1] then return false end
    end
    return true
end

function shuffle(arr)
    for i = #arr, 2, -1 do
        local j = math.random(i)
        arr[i], arr[j] = arr[j], arr[i]
    end
end

function bogo_sort(arr)
    while not is_sorted(arr) do
        shuffle(arr)
    end
end

local arr = {3, 1, 4, 2}
bogo_sort(arr)
print(table.concat(arr, " "))

File: JavaScript/bogo_sort.js
function bogoSort(arr) {
    function isSorted(arr) {
        for (let i = 1; i < arr.length; i++) {
            if (arr[i - 1] > arr[i]) {
                return false;
            }
        }
        return true;
    }

    function shuffle(arr) {
        for (let i = arr.length - 1; i > 0; i--) {
            const j = Math.floor(Math.random() * (i + 1));
            [arr[i], arr[j]] = [arr[j], arr[i]];
        }
    }

    while (!isSorted(arr)) {
        shuffle(arr);
    }

    return arr;
}

const array = [3, 2, 5, 1, 4, 0];
const sortedArray = bogoSort(array);
console.log("Sorted array:", sortedArray);